src/createApiCall.ts

```diff
@@ -89,23 +100,38 @@ export function createApiCall(
 
         const streaming = (currentApiCaller as StreamingApiCaller).descriptor
           ?.streaming;
+
         const retry = thisSettings.retry;
-        if (
-          !streaming &&
-          retry &&
-          retry.retryCodes &&
-          retry.retryCodes.length > 0
-        ) {
-          retry.backoffSettings.initialRpcTimeoutMillis =
-            retry.backoffSettings.initialRpcTimeoutMillis ||
-            thisSettings.timeout;
-          return retryable(
-            func,
-            thisSettings.retry!,
-            thisSettings.otherArgs as GRPCCallOtherArgs,
-            thisSettings.apiName
+        if (!streaming && retry && retry.getResumptionRequestFn) {
+          throw new Error(
+            'Resumption strategy can only be used with server streaming retries'
           );
         }
+        if (!streaming && retry && retry.retryCodesOrShouldRetryFn) {
+          if (
+            retry.retryCodesOrShouldRetryFn instanceof Array &&
+            retry.retryCodesOrShouldRetryFn.length > 0
+          ) {
+            retry.backoffSettings.initialRpcTimeoutMillis =
+              retry.backoffSettings.initialRpcTimeoutMillis ||
```

Comment:
Please use `??` instead of `||` (makes a difference if `initialRpcTimeoutMillis` is 0)

Shortened, this would be `retry.backoffSettings.initialRpcTimeoutMillis ??= thisSettings.timeout;`

Reply:
great catch